radix-engine/src/kernel/interpreters/scrypto_interpreter.rs

```diff
@@ -141,29 +129,15 @@ impl ExecutableInvocation for ScryptoInvocation {
 
         let export_name = match type_info {
             TypeInfoSubstate::NativePackage => {
-                let package_global = RENodeId::Global(GlobalAddress::Package(self.package_address));
-                let handle = api.lock_substate(
-                    package_global,
-                    NodeModuleId::SELF,
-                    SubstateOffset::Package(PackageOffset::Info),
-                    LockFlags::read_only(),
-                )?;
-                let substate_ref = api.get_ref(handle)?;
-                let package = substate_ref.native_package_info();
-                for dependent_resource in &package.dependent_resources {
-                    node_refs_to_copy.insert(RENodeId::Global(GlobalAddress::Resource(
-                        *dependent_resource,
-                    )));
-                }
-
-                api.drop_lock(handle)?;
-
+                // TODO: Do we need to check against the abi? Probably not since we should be able to verify this
+                // TODO: in the native package itself.
```

Comment:
Haha, are we confident enough about ourselves? May help identify inconsistency between the stored ABIs and actual impl.

Reply:
This is not that true for the check on the Input since we always deserialize to the Input struct type. If `ScryptoValue -> RustStruct` wasn't so expensive this would be good.